address_book/internal/service/service.go

```diff
@@ -35,44 +62,24 @@ func (a *AddressBookService) FindContact(_ context.Context, in *pb.FindContactRe
 		}, nil
 	}
 
-	contacts := []*pb.Contact{}
-
 	switch in.SearchType {
 	case pb.FindContactRequest_NAME:
-		a.data.Range(func(key interface{}, value interface{}) bool {
-			if contact, ok := value.(*pb.Contact); ok {
-				if contact.Name == in.Query {
-					contacts = append(contacts, contact)
-				}
-			}
-			return true
-		})
+		findContacts := []models.Contact{}
+
+		err := a.db.Where("name = ?", in.Query).Find(&findContacts).Error
+
+		return processFindContact(&findContacts, err)
 	case pb.FindContactRequest_PHONE:
-		matchString := in.Query
-
-		a.data.Range(func(key interface{}, value interface{}) bool {
-			if contact, ok := value.(*pb.Contact); ok {
-				matchString := strings.ReplaceAll(matchString, "?", ".")
-				matched, err := regexp.MatchString(matchString, contact.Phone)
-
-				if err == nil {
-					if matched {
-						contacts = append(contacts, contact)
-					}
-				}
-			}
-			return true
-		})
+		findContacts := []models.Contact{}
+
+		err := a.db.Where("phone LIKE ?", in.Query).Find(&findContacts).Error
```

Comment:
LIKE syntax differs from wildcard syntax, you need to format it.

Reply:
Please specify which implementation you require, the wildcards are different for different databases (For example: "?', "*" for MS Access and "_", "%" for Postgres)

Reply:
I mean the same syntax that you supported in the task 1. Please keep it consistent.

Reply:
Done